internal/util/pki/ca.go

```diff
@@ -82,6 +82,14 @@ func WithAuthority(authority string) CSROption {
 	})
 }
 
+// WithSAN adds the given host or IP to the certificates Subject Alternative Names list.
+// As a convenience, if the hostOrIP has a port it will be stripped before adding.
```

Comment:
I would prefer to see this separated out.
Have the caller do `WithSAN(StripPort(foo))` instead of adding this extra logic to `hydrateTemplate`

Reply:
That would be a breaking change as the existing authority logic strips the ports. Well maybe not breaking, but introducing inconsistencies.

Reply:
Fair enough, keep it as-is